### PR TITLE
Migrate from jenkins to GitHub actions

### PR DESCRIPTION
**Pull request by joakimen**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-          npm run snapshots
-          git status
-          git diff --exit-code
-        '''
-      }
-
-      // We only run semantic-release on the release branches,
-      // as we do not want credentials to be exposed to the job
-      // on other branches or in PRs.
-      if (env.BRANCH_NAME ==~ /^(master|\d+\.(\d+|x)(\.x)?)$/) {
```
Comment (joakimen) [inline Jenkinsfile:36]: Didn't include support for releasing from non-trunk branches, such as referenced here.
We don't have any such branches today, nor any plans on supporting it afaik

**Comment by joakimen** [comment]
Before adding these changes, we need to decide how to handle public NPM packages, marking as draft for now.

**Comment by joakimen** [comment]
> Before adding these changes, we need to decide how to handle public NPM packages, marking as draft for now.

npm token strategy has been decided, and token has been added to workflow according to [the semrel guide](https://semantic-release.gitbook.io/semantic-release/recipes/ci-configurations/github-actions)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+# NOTE: Set concurrency for the current workflow to 1
+concurrency: ci-${{ github.ref }}-${{ github.workflow }}
+
+jobs:
+  build-and-deploy:
+    timeout-minutes: 60
+    runs-on: ubuntu-22.04
+    permissions:
+      actions: read
```
Comment (stekern) [inline .github/workflows/ci.yml:21]: Not required by the job

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      contents: write
+      id-token: write
```
Comment (stekern) [inline .github/workflows/ci.yml:23]: Not required by the job

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+# NOTE: Set concurrency for the current workflow to 1
+concurrency: ci-${{ github.ref }}-${{ github.workflow }}
+
+jobs:
+  build-and-deploy:
+    timeout-minutes: 60
+    runs-on: ubuntu-22.04
+    permissions:
```
Comment (stekern) [inline .github/workflows/ci.yml:20]: We likely also need `pull-requests: write` and `issues: write` permissions (https://github.com/semantic-release/semantic-release/issues/2469#issuecomment-1158013884)

**Review by stekern — CHANGES_REQUESTED** [review]
A couple of comments regarding job permissions, but otherwise looks good 🚀

File: .github/workflows/ci.yml (hunk 3 above)
Comment (joakimen) [reply .github/workflows/ci.yml]: https://github.com/capralifecycle/liflig-cdk/pull/231/commits/233b66cebbae498098c6309fde8a3f0c4def998d

File: .github/workflows/ci.yml (hunk 4 above)
Comment (joakimen) [reply .github/workflows/ci.yml:20]: https://github.com/capralifecycle/liflig-cdk/pull/231/commits/a03e47063c838d082ca55497ac0e5869d4d842df 
https://github.com/capralifecycle/liflig-cdk/pull/231/commits/1ce9bccf8d9cde660781ddde7875b539cb2ab183

File: .github/workflows/ci.yml (hunk 2 above)
Comment (joakimen) [reply .github/workflows/ci.yml]: https://github.com/capralifecycle/liflig-cdk/pull/231/commits/233b66cebbae498098c6309fde8a3f0c4def998d

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  run:
+    # NOTE: A bit stricter than the default bash options used by GitHub Actions
+    # (bash --noprofile --norc -e -o pipefail {0})
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+# NOTE: Set concurrency for the current workflow to 1
+concurrency: ci-${{ github.ref }}-${{ github.workflow }}
+
+jobs:
+  build-and-deploy:
```
Comment (stekern) [inline .github/workflows/ci.yml:17]: nit: build-and-release

**Review by stekern — APPROVED** [review]
LGTM!

I suggest we clean up the commit history a bit, and use a conventional commit type (e.g., `fix(ci): use github actions` or `feat(ci): use github actions`) that will trigger a release so we can verify that everything works as expected upon merge.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (joakimen) [reply .github/workflows/ci.yml]: https://github.com/capralifecycle/liflig-cdk/pull/231/commits/c1fb632ba1de7fd498f36615a2d2b903c9f1740d#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR17

**Pull request merged**